obs/project.go

```diff
@@ -40,13 +40,98 @@ type Project struct {
 	UseForBuild  *UseForBuild `json:"useForBuild,omitempty" xml:"useforbuild,omitempty"`
 }
 
-type Client struct {
-	Client   http.Client
+// OBS is the wrapper struct
+type OBS struct {
+	client  Client
+	options *Options
+}
+
+type obsClient struct {
+	*http.Client
+	*Options
+}
+
+// InvokeOBSEndpoint is the Interface method implementation over obsClient
+func (c *obsClient) InvokeOBSEndpoint(ctx context.Context, method string, url string, xmlData *bytes.Buffer) (*http.Response, error) {
+	//TODO implement me
+	req, err := http.NewRequestWithContext(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: creating request: %v", err),
+		}
+	}
+
+	req.SetBasicAuth(c.Username, c.Password)
+	req.Header.Set("Accept", "application/xml; charset=utf-8")
+
+	resp, err := c.InvokeOBSEndpoint(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: sending request: %v", err),
+		}
+	}
+
+	return resp, nil
+}
+
+type Client interface {
+	InvokeOBSEndpoint(ctx context.Context, method string, url string, xml *bytes.Buffer) (*http.Response, error)
+}
+
+func (o *OBS) InvokeOBSEndpoint(ctx context.Context, method string, url string, xmlData *bytes.Buffer) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: creating request: %v", err),
+		}
+	}
+
+	req.SetBasicAuth(o.options.Username, o.options.Password)
+	req.Header.Set("Accept", "application/xml; charset=utf-8")
+
+	resp, err := o.InvokeOBSEndpoint(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: sending request: %v", err),
+		}
+	}
+
+	return resp, nil
+}
```

Comment:
This is not needed, please remove it

Reply:
Then how are we going to define the body of this function especially during testing?

Reply:
@nitishfy This comment is still not fixed